### PR TITLE
JNG-5602 Add JSL project generator templat (crash on frontend generat…

### PR DESCRIPTION
**Pull request by robertcsakany**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5602" title="JNG-5602" target="_blank">Sub-taskJNG-5602</a>  Add JSL Fullstack Karaf template plugin
  </td></table>
  <br />
 

…ion)

### CODE REVIEW

**Review by gaborflorian — APPROVED** [review]

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        <frontend-type>react</frontend-type>
+
+        <!-- React -->
+        <node-version-defined>18.16.0</node-version-defined>
+        <pnpm-version-defined>8.6.1</pnpm-version-defined>
+        <judo-meta-ui-version-defined>1.0.0.20220618_232754_311274d9_develop</judo-meta-ui-version-defined>
+        <judo-ui-archetype-version-defined>1.0.0.20221003_142712_453def5b_develop</judo-ui-archetype-version-defined>
+        <judo-tatami-ui2web-version-defined>1.0.0.20230117_190258_43070f4d_develop</judo-tatami-ui2web-version-defined>
+        <judo-tatami-ui2typescript-maven-plugin-version-defined>1.0.0.20230117_190258_43070f4d_develop</judo-tatami-ui2typescript-maven-plugin-version-defined>
+        <judo-tatami-ui-external-packages-version-defined>1.0.0.20230117_190258_43070f4d_develop</judo-tatami-ui-external-packages-version-defined>
```
Comment (noherczeg) [inline README.adoc:52]: I don't think we need this any more.

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/README.adoc.hbs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+./judo.sh build
+----
+
+
+=== Start the application
+After build it can be started
+
+[source,]
+----
+./judo.sh build
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/README.adoc.hbs]: build?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/compose-develop/docker-compose.yml.hbs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    image: {{ lowerCase model.name }}-application:latest
+    labels:
+      - "traefik.http.routers.app.rule=Host(`app.traefik.me`)"
+      - "traefik.http.services.app.loadbalancer.server.port=8181"
+    environment:
+      - EXTRA_JAVA_OPTS=-Xms1024m -Xmx1024m -Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8
+      - JUDO_PLATFORM_IDENTIFIER_SIGNER_SECRET=tJ0lt0h9X2WvVydp4SoDIahBdBcctMxlTn6UJ1xTbQYltBF//qPLmyZPU6X02ETnsP/X7G3IDhGttNXISwyi1uJk5haBj3MI88baDxckmOKpAd6Fy93KSZJgPoGuICD0bPYKeXMSSqEKypt4NkY0NiNL2aLVwL3UDCWHsWQsCfk=
+      - JUDO_PLATFORM_KEYCLOAK_DEFAULT_PASSWORD_POLICY=SAME_USERNAME
+      - JUDO_PLATFORM_LOG_SMTP_SERVER=true
+      - JUDO_PLATFORM_FILESTORE=rdbms
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/compose-develop/docker-compose.yml.hbs:37]: This is not recommended, not a best practice.

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/compose-develop/docker-compose.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  app:
+    image: {{ lowerCase model.name }}-application:latest
+    labels:
+      - "traefik.http.routers.app.rule=Host(`app.traefik.me`)"
+      - "traefik.http.services.app.loadbalancer.server.port=8181"
+    environment:
+      - EXTRA_JAVA_OPTS=-Xms1024m -Xmx1024m -Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8
+      - JUDO_PLATFORM_IDENTIFIER_SIGNER_SECRET=tJ0lt0h9X2WvVydp4SoDIahBdBcctMxlTn6UJ1xTbQYltBF//qPLmyZPU6X02ETnsP/X7G3IDhGttNXISwyi1uJk5haBj3MI88baDxckmOKpAd6Fy93KSZJgPoGuICD0bPYKeXMSSqEKypt4NkY0NiNL2aLVwL3UDCWHsWQsCfk=
+      - JUDO_PLATFORM_KEYCLOAK_DEFAULT_PASSWORD_POLICY=SAME_USERNAME
+      - JUDO_PLATFORM_LOG_SMTP_SERVER=true
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/compose-develop/docker-compose.yml.hbs:36]: Do we need this in a simple app?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+                            <execution>
+                                <id>build-platform-docker-x86_64</id>
+                                <phase>package</phase>
+                                <goals>
+                                    <goal>buildTar</goal>
+                                </goals>
+
+                                <configuration>
+                                    <from>
+                                        <image>azul/zulu-openjdk-debian:17-latest</image> -->
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs:243]: Not specifying a version number results in different images being built every single time. Which means we will never be able to tell what version are we compatible with. 

Version should at least be a property.

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+                             <execution>
+                                 <id>build-platform-docker-amd64</id>
+                                 <phase>package</phase>
+                                 <goals>
+                                     <goal>buildTar</goal>
+                                 </goals>
+
+                                 <configuration>
+                                     <from>
+                                         <image>azul/zulu-openjdk-debian:17-latest</image> -->
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs:348]: Same here

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+                            <execution>
+                                <id>build-platform-docker-arm64</id>
+                                <phase>package</phase>
+                                <goals>
+                                    <goal>buildTar</goal>
+                                </goals>
+
+                                <configuration>
+                                    <from>
+                                        <image>azul/zulu-openjdk-debian:17-latest</image>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/docker/pom.xml.hbs:453]: And here

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/frontend-react/model/pom.xml.hbs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    {{> frontend-react/model/pom.xml.project-definition.fragment.hbs }}
+
+    <packaging>pom</packaging>
+
+    <build>
+        <plugins>
+        <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/frontend-react/model/pom.xml.hbs]: Do we need this?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/frontend-react/model/pom.xml.hbs [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+                                    <outputDirectory>${project.basedir}/</outputDirectory>
+                                    <includes>**/*-ui.model</includes>
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/frontend-react/model/pom.xml.hbs]: Or this?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/internal/pom.xml.hbs [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+    <parent>
+        <artifactId>{{ lowerCase model.name }}{{ projectPostfix }}</artifactId>
+        <groupId>{{ groupId }}</groupId>
+        <version>${revision}</version>
+    </parent>
+
+    <artifactId>{{ lowerCase model.name }}{{ projectPostfix }}-internal</artifactId>
+    <name>{{ model.name }} SDK Internal</name>
+    <description>{{ model.name }} - SDK Internal is the bridge the implemented app functions and runtime</description>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/internal/pom.xml.hbs]: bridge between?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/model/pom.xml.hbs [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    <artifactId>{{ lowerCase model.name }}{{ projectPostfix }}-model</artifactId>
+
+    <name>{{ model.name }} transformed models</name>
+    <description>{{ model.name }} transformed models</description>
+
+    <packaging>bundle</packaging>
+
+    <build>
+        <plugins>
+            <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/model/pom.xml.hbs]: ?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/model/pom.xml.hbs [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+                        <version>${judo-tatami-jsl-version}</version>
+                        <executions>
+                            <execution>
+                                <id>transform-jsl-file</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>default-model-workflow</goal>
+                                </goals>
+                                <configuration>
+                                    <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/model/pom.xml.hbs]: ?

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/rest/pom.xml.hbs [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+            <plugin>
+                <groupId>io.swagger.core.v3</groupId>
+                <artifactId>swagger-maven-plugin</artifactId>
+                <version>2.2.19</version>
+                <executions>
+                    <execution>
+                        <configuration>
+                            <outputFileName>openapi</outputFileName>
+                            <outputPath>${project.build.directory}/api-documentation-swagger</outputPath>
+                            <outputFormat>JSONANDYAML</outputFormat>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/rest/pom.xml.hbs:91]: Do we need both? Would it improve the generator performance if we only generated one of these?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/rest/pom.xml.hbs [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>{{ lowerCase model.name }}{{ projectPostfix }}</artifactId>
+        <groupId>{{ groupId }}</groupId>
+        <version>${revision}</version>
+    </parent>
+
+    <artifactId>{{ lowerCase model.name }}{{ projectPostfix }}-rest</artifactId>
+    <name>{{ model.name }} Jakarta RESTful Web Services</name>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/rest/pom.xml.hbs:13]: There is no dependency / module under this one with the Jakarta namespace, only CXF. Is this name correct?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/schema/README.adoc.hbs [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+----
+
+Generate migration scripts:
+
+[source,bash]
+----
+mvn judo-rdbms-schema:generate -DbaseModelVersion=1.1.0.20210929_004050_9602796_develop -DupdateModelVersion=1.1.0.20210929_154229_0d13206_develop
+----
+
+Execute migration with the something like following command:
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/schema/README.adoc.hbs:45]: "Execute the migration directly"?

File: judo-jsl-fullstack-karaf-project-template-root/src/main/resources/mvn/jvm.config [hunk 16]
```diff
@@ -0,0 +1 @@
+-Dfile.encoding=UTF-8 --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.time=ALL-UNNAMED -Dtycho.disableP2Mirrors=true -Djansi.force=true
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-root/src/main/resources/mvn/jvm.config]: Missing new line

File: judo-jsl-fullstack-karaf-project-template-test/pom.xml [hunk 17]
```diff
@@ -0,0 +1,382 @@
...
+            <id>generatorTest</id>
+            <activation>
+                <property>
+                    <name>skipGeneratorTest</name>
+                    <value>!true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-test/pom.xml]: ?

File: judo-jsl-fullstack-karaf-project-template-test/pom.xml [hunk 18]
```diff
@@ -0,0 +1,382 @@
...
+                                <javaCompiler>ecj</javaCompiler>
+
+                            </templateParameters>
+                        </configuration>
+                    </execution>
+                </executions>
+
+            </plugin>
+
+            <!--
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-test/pom.xml:172]: ?

File: pom.xml [hunk 19]
```diff
@@ -0,0 +1,722 @@
...
+        <judo-version-updater-maven-plugin-version>1.0.0.20231027_121339_45b7ebd2_develop</judo-version-updater-maven-plugin-version>
+
+        <surefire-version>3.0.0-M7</surefire-version>
+        <!--suppress UnresolvedMavenProperty -->
+        <logback-test-config>${maven.multiModuleProjectDirectory}/logback-test.xml</logback-test-config>
+
+        <!-- Code Quality-->
+        <sonar-maven-plugin-version>3.9.1.2184</sonar-maven-plugin-version>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
```
Comment (noherczeg) [inline pom.xml:53]: Do we want to mess with Sonar still?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs [hunk 20]
```diff
@@ -0,0 +1,406 @@
...
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.11.0</version>
+                    {{# eq javaCompiler "ecj"}}
+                    <configuration>
+                        <compilerId>eclipse</compilerId>
+                        <source>17</source>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs:42]: Shouldn't this come from properties? Maybe from a parent?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs [hunk 21]
```diff
@@ -0,0 +1,406 @@
...
+                        <id>flatten.clean</id>
+                        <phase>clean</phase>
+                        <goals>
+                            <goal>clean</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- Report active profiles on console. -->
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs]: ?

File: judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs [hunk 22]
```diff
@@ -0,0 +1,406 @@
...
+                <property>
+                    <name>!skipSDK</name>
+                </property>
+            </activation>
+
+            <modules>
+                <module>sdk</module>
+                <module>internal</module>
+                <module>app</module>
+                <module>rest</module>
```
Comment (noherczeg) [inline judo-jsl-fullstack-karaf-project-template-application/src/main/resources/pom.xml.hbs:241]: Why is rest part of sdk?

**Review by noherczeg — CHANGES_REQUESTED** [review]

**Review by noherczeg — APPROVED** [review]

**Pull request merged**